src/types/context.ts

```diff
@@ -5,82 +5,78 @@ export enum ShopContext {
 }
 
 export interface BackendUser {
-  email?: string | null;
-  employeeId?: number;
-  isSuperAdmin: boolean;
+  email: string | null
+  employeeId: number
+  isSuperAdmin: boolean
 }
 
-export type Context = {
-  accountsUiUrl?: string | null;
-  adminAjaxLink?: string | null;
-  backendUser?: Partial<BackendUser>;
-  currentContext?: Partial<CurrentContext>;
-  currentShop?: Shop | null;
-  dependencies?: Record<
-    string,
-    {
-      isInstalled?: boolean;
-      isEnabled?: boolean;
-      installLink?: string;
-      enableLink?: string;
-    }
-  >;
-  isOnboardedV4?: boolean;
-  isShopContext?: boolean;
-  manageAccountLink?: string | null;
-  onboardingLink?: string | null;
-  psAccountsEnableLink?: string | null;
-  psAccountsInstallLink?: string | null;
-  psAccountsIsEnabled?: boolean;
-  psAccountsIsInstalled?: boolean;
-  psAccountsIsUptodate?: boolean;
-  psAccountsUpdateLink?: string | null;
-  psAccountsVersion?: string;
-  psIs17: boolean;
-  psxName?: string;
-  shops: ShopGroup[];
-  ssoResendVerificationEmail?: string | null;
-  superAdminEmail?: string | null;
-  user?: Partial<User> | null;
-  errors?: string[];
-};
+export interface Context {
```

Comment:
Maybe Joi doesn't support type inference, but [Zod](https://zod.dev/) seems to [be able to do this](https://www.allthingstypescript.dev/p/using-zod-schemas-as-source-of-truth).

Reply:
Yes but for zod i think it's the other way you can generate a type from the validation schema, and not passing an interface to generate the validation schema

Reply:
With Joi, we didn't pass an interface to generate the validation schema as we use Joi methods.

From what I understood of Zod, we use equivalent methods to validate the data, but we can get a type from the validation schema. We will still have `src/lib/ContextValidator.ts` file but `src/types/context.ts` would be provided by Zod